tests/languages/flutter/tests.dart

```diff
@@ -1,10 +1,21 @@
 import 'package:flutter/material.dart';
 import '../lib/packageName.dart';
+import '../lib/client_io.dart';
 import '../lib/models.dart';
+import 'dart:io';
+
+class MockClient extends ClientIO {
+  Future<Directory> getCookiePath() async {
```

Comment:
Is this done because of path_provider? Perhaps we should be replacing the [PathProviderPlatform.instance](https://github.com/flutter/plugins/blob/8fcff8756c9888ac112afa12baacbb186b07f982/packages/path_provider/path_provider/test/path_provider_test.dart#LL24C52-L24C52) instead?

Reply:
Yes, but the suggestion was to create a mocked [PathProviderPlatform.instance](https://github.com/flutter/plugins/blob/8fcff8756c9888ac112afa12baacbb186b07f982/packages/path_provider/path_provider/test/path_provider_test.dart#LL24C52-L24C52).

Reply:
Only mocking might not work, because we need actual directory, as we have cookie test, that requires actually saving the cookie in cache, Otherwise, I'll have to mock cookie jar and other cookie related stuff including the interceptors

Reply:
@lohanidamodar, sorry, I don't quite understand the problem. What's the problem with mocking [PathProviderPlatform.instance](https://github.com/flutter/plugins/blob/8fcff8756c9888ac112afa12baacbb186b07f982/packages/path_provider/path_provider/test/path_provider_test.dart#LL24C52-L24C52) and having `getApplicationDocumentsDirectory()` return `.`. Then, the following

```dart
    final directory = await getApplicationDocumentsDirectory();
    final path = directory.path;
    final Directory dir = Directory('$path/cookies');
    await dir.create();
    return dir;
```

would be the same as what you have here, right?

Reply:
Updated!